jax/interpreters/pxla.py

```diff
@@ -1111,8 +1108,8 @@ def _pmap_translation_rule(c, axis_env,
                      "or another pmap is not supported.")
   # We in-line here rather than generating a Call HLO as in the xla_call
   # translation rule just because the extra tuple stuff is a pain.
-  if axis_env.devices is not None or (axis_env.names and devices is not None):
```

Comment:
Missing context - why is it ok to get rid of the axis_env.devices check here?

Reply:
axis_env.devices are devices from any outer pmaps. So that's ok now (since the outermost pmap can have devices, even if there's an inner pmap), but we still want to error if there are devices on the inner pmap (the remaining check).

Reply:
makes sense, LGTM once tests pass.